pkg/vault/contracts/CompositeLiquidityRouter.sol

```diff
@@ -258,21 +284,23 @@ contract CompositeLiquidityRouter is ICompositeLiquidityRouter, BatchRouterCommo
             })
         );
 
-        (underlyingAmountsIn, ) = _wrapTokens(
-            params,
+        amountsIn = _wrapTokensExactOutIfRequired(
+            params.sender,
+            useWrappedTokens,
             erc4626PoolTokens,
             wrappedAmountsIn,
-            SwapKind.EXACT_OUT,
-            params.maxAmountsIn
+            params.maxAmountsIn,
+            params.wethIsEth
         );
     }
 
     function removeLiquidityERC4626PoolProportionalHook(
-        RemoveLiquidityHookParams calldata params
-    ) external nonReentrant onlyVault returns (uint256[] memory underlyingAmountsOut) {
+        RemoveLiquidityHookParams calldata params,
+        bool[] calldata useWrappedTokens
+    ) external nonReentrant onlyVault returns (uint256[] memory amountsOut) {
         IERC20[] memory erc4626PoolTokens = _vault.getPoolTokens(params.pool);
         uint256 poolTokensLength = erc4626PoolTokens.length;
-        underlyingAmountsOut = new uint256[](poolTokensLength);
+        amountsOut = new uint256[](poolTokensLength);
 
```

Comment:
// Revert if `tokensOut` length does not match `minAmountsOut` and `useWrappedTokens`.
        InputHelpers.ensureInputLengthMatch(poolTokensLength, params.minAmountsOut.length, useWrappedTokens.length);